public/app/features/alerting/unified/api/ruler.ts

```diff
@@ -155,8 +156,13 @@ async function rulerGetRequest<T>(url: string, empty: T, params?: Record<string,
 }
 
 function isResponseError(error: unknown): error is FetchResponse<ErrorResponseMessage> {
-  const hasErrorMessage = (error as FetchResponse<ErrorResponseMessage>).data != null;
-  const hasErrorCode = Number.isFinite((error as FetchResponse<ErrorResponseMessage>).status);
+  if (!isObject(error)) {
+    return false;
+  }
+
+  const hasErrorMessage = 'data' in error && error.data !== null;
```

Comment:
Was the change to strict equality here intentional?

Reply:
Alrighty, can we change it to `error.data !== undefined && error.data !== null` then to keep the same behaviour as before?